Makefile

```diff
@@ -19,8 +19,12 @@ ALL_PKGS := $(shell go list $(sort $(dir $(ALL_SRC))))
 
 ALL_TESTS_DIRS := $(shell find tests -name '*_test.go' | xargs -L 1 dirname | uniq | sort -r)
 
+ALL_EXAMPLES_DOCKER_COMPOSE_TESTS := $(shell find examples -name 'docker-compose.yml' |  xargs dirname | xargs basename | awk '{print "example-"$$1}')
```

Comment:
`examples/CONTRIBUTING.md` makes sense to me since I'm under the impression `docker-compose up` would be the only necessary command for the example to run. Separating dep/testing logic from user workflow is preferred and contributor relevance helps w/ that.

@hughesjj I'm not sure I understand the relation of your suggestion to the examples updates in these changes. I think it's unlikely someone would need to run all make targets in the project makefile but if you have improvements in mind feel free to open a PR or a planning issue.